magenta-lib/src/main/scala/magenta/Project.scala

```diff
@@ -130,14 +131,18 @@ object HostList {
   implicit def hostListAsListOfHosts(hostList: HostList) = hostList.hosts
 }
 
+case class RuntimeResources(reporter: DeployReporter, lookup: Lookup, parameters: DeployParameters, stack: Stack) {
```

Comment:
I'm not keen on the name and feel that partly that's because there are two different types of thing here. 

To me, the reporter and the lookup are things a task might need to perform an action, whereas the params and stack determine what the deployment tries to do.